aliases/available/general.aliases.bash

```diff
@@ -2,19 +2,55 @@ cite about-alias
 about-alias 'general aliases'
 
 # List directory contents
-alias sl=ls
-alias ls='ls -G'        # Compact view, show colors
-alias la='ls -AF'       # Compact view, show hidden
-alias ll='ls -al'
-alias l='ls -a'
-alias l1='ls -1'
+# Alias source: http://tldp.org/LDP/abs/html/sample-bashrc.html
+
+local LS_COLOR_OPTION
+
+case "$OSTYPE" in
+  linux*)
+    # Set Linux color option
+    LS_COLOR_OPTION="--color=auto"
+    ;;
+  darwin*)
+    # Set BSD color optio
+    LS_COLOR_OPTION="-G"
+    #Check if coreutils version of if exists
+    if [[ -x "/usr/local/opt/coreutils/libexec/gnubin/ls" ]]; then
```

Comment:
I don't think we should use this - we shouldn't rely on a specific path.

The [solution](http://superuser.com/a/879035/119764) I had posted in my comment uses the following construct to verify whether `ls` supports the `--color` option - without checking for a path entry:

```bash
export CLICOLOR=1

ls --color=auto &> /dev/null && alias ls='ls --color=auto'
```

This first sets the `CLICOLOR` variable, which will enable colored output for the stock `ls` version, and then tries to run `ls --color=auto` to check whether it fails or not. If it works, it defines the alias to use `--color=auto`.

You could do the same check here, and then set the `LS_COLOR_OPTION` accordingly.